### PR TITLE
additional functions to support builds

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

File: vars/projectExists.groovy [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/groovy
+
+def call(Object ctx, String projectName, String token) {
+  def exists = false
+  ctx.openshift.withCluster() {
+    try {
+      if (token) {
+        ctx.openshift.doAs(token) {
```
Comment (stevekuznetsov) [inline vars/projectExists.groovy:8]: Why?

Comment (csrwng) [reply vars/projectExists.groovy:8]: We want to allow you to run this particular pipeline in your own machine or on a cluster that you have access to. Not everyone has access to give the jenkins sa the self-provisioner role which is what you need to check project existence or create new projects. For that case, you can pass your own token in, and we'll use it. For origin-ci, you already granted the jenkins sa the self provisioner role so we should be good without passing one.

Comment (stevekuznetsov) [reply vars/projectExists.groovy:8]: Makes sense.

**Comment by stevekuznetsov** [comment]
@kargakis wtf?

**Comment by stevekuznetsov** [comment]
@kargakis I can't even see the webhooks for this hitting our `hook`

**Comment by 0xmichalis** [comment]
Turn on debug

**Comment by stevekuznetsov** [comment]
Hmmmm yeah ok next week I will tackle logging. That is such an integral part of hook it makes no sense to need debug for it

**Pull request merged**